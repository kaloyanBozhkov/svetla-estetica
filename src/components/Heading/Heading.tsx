import Button from 'components/UI/Button/Button'
import styles from './heading.module.scss'

type HeadingProps = {
    label: string
    buttonProps?: {
        label: string
        onClick: () => void
    }
}

const Heading = ({ label, buttonProps }: HeadingProps): JSX.Element => {
    return (
        <div className={styles.heading}>
            <p>{label}</p>
            {buttonProps && (
                <div className={styles.btnWrapper}>
                    <Button {...buttonProps} modifier="solid" />
                </div>
            )}
        </div>
    )
}

export default Heading
