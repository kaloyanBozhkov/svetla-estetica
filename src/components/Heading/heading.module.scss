@import "scss/variables.scss";
@import "scss/mixins.scss";

.heading {
    width: 100%;
    height: 8.8rem;
    background-color: $white;
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    
    @include sidePaddings;
    @include slyBoxShadow;

    > p {
        font-size: 2.5rem;
        color: $black;
    }

    .btnWrapper {
        height: 3.2rem;
        width: 15.6rem;
    }
}
