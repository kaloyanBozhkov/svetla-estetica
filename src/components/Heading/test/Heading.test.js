import React from 'react'
import { shallow } from 'enzyme'
import Heading from '../Heading'

describe('testing SeactionHeading', () => {
    it('should render component', () => {
        expect(shallow(<Heading label="Some Section Label" />)).toMatchSnapshot()
    })

    it('should render heading with an action btn', () => {
        const mockClick = jest.fn(),
            wrapper = shallow(
                <Heading buttonProps={{ label: 'Some label', onClick: mockClick }} />
            ),
            btn = wrapper.find('Button')

        expect(btn.length).toBe(1)
        btn.prop('onClick')()
        expect(mockClick).toHaveBeenCalled()
    })
})
