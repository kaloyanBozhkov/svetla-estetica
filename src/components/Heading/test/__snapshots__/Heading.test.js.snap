// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SeactionHeading should render component 1`] = `
<div
  className="heading"
>
  <p>
    Some Section Label
  </p>
</div>
`;
